dev/bots/unpublish_package.dart

```diff
@@ -0,0 +1,525 @@
+  final ProcessResult result;
+  int get exitCode => result?.exitCode ?? -1;
+
+  @override
+  String toString() {
+    String output = runtimeType.toString();
+    if (message != null) {
+      output += ': $message';
+    }
+    final String stderr = result?.stderr ?? '';
+    if (stderr.isNotEmpty) {
+      output += ':\n$stderr';
+    }
+    return output;
+  }
+}
+
+enum Channel { dev, beta, stable }
+
+String getChannelName(Channel channel) {
+  switch (channel) {
+    case Channel.beta:
+      return 'beta';
+    case Channel.dev:
+      return 'dev';
+    case Channel.stable:
+      return 'stable';
+  }
+  return null;
+}
+
+Channel fromChannelName(String name) {
+  switch (name) {
+    case 'beta':
+      return Channel.beta;
+    case 'dev':
+      return Channel.dev;
+    case 'stable':
+      return Channel.stable;
+    default:
+      throw ArgumentError('Invalid channel name.');
+  }
+}
+
+enum PublishedPlatform { linux, macos, windows }
+
+String getPublishedPlatform(PublishedPlatform platform) {
+  switch (platform) {
+    case PublishedPlatform.linux:
+      return 'linux';
+    case PublishedPlatform.macos:
+      return 'macos';
+    case PublishedPlatform.windows:
+      return 'windows';
+  }
+  return null;
+}
+
+PublishedPlatform fromPublishedPlatform(String name) {
+  switch (name) {
+    case 'linux':
+      return PublishedPlatform.linux;
+    case 'macos':
+      return PublishedPlatform.macos;
+    case 'windows':
+      return PublishedPlatform.windows;
+    default:
+      throw ArgumentError('Invalid published platform name.');
+  }
+}
+
+/// A helper class for classes that want to run a process, optionally have the
+/// stderr and stdout reported as the process runs, and capture the stdout
+/// properly without dropping any.
+class ProcessRunner {
+  ProcessRunner({
+    ProcessManager processManager,
+    this.subprocessOutput = true,
+    this.defaultWorkingDirectory,
+    this.platform = const LocalPlatform(),
+  }) : processManager = processManager ?? const LocalProcessManager() {
+    environment = Map<String, String>.from(platform.environment);
+  }
+
+  /// The platform to use for a starting environment.
+  final Platform platform;
+
+  /// Set [subprocessOutput] to show output as processes run. Stdout from the
+  /// process will be printed to stdout, and stderr printed to stderr.
+  final bool subprocessOutput;
+
+  /// Set the [processManager] in order to inject a test instance to perform
+  /// testing.
+  final ProcessManager processManager;
+
+  /// Sets the default directory used when `workingDirectory` is not specified
+  /// to [runProcess].
+  final Directory defaultWorkingDirectory;
+
+  /// The environment to run processes with.
+  Map<String, String> environment;
+
+  /// Run the command and arguments in `commandLine` as a sub-process from
+  /// `workingDirectory` if set, or the [defaultWorkingDirectory] if not. Uses
+  /// [Directory.current] if [defaultWorkingDirectory] is not set.
```

Comment:
You could go whole hog with the DI if you wanted and have a `final FileSystem fileSystem` instance variable, then use `fileSystem.currentDirectory`...

It only really matters if you add tests for this.